internal/db/error_codes.go

```diff
@@ -0,0 +1,17 @@
+package db
+
+import (
+	"errors"
+
+	"github.com/jackc/pgconn"
+)
+
+// https://www.postgresql.org/docs/current/errcodes-appendix.html
+
+func IsUniqueConstraintError(err error) error {
+	var pgErr *pgconn.PgError
+	if err != nil && errors.As(err, &pgErr) && pgErr.Code == "23505" {
+		return pgErr
+	}
+	return nil
+}
```

Comment:
Let's put all postgres error code detection here so we can utilize this for meaningful error messages.